ocp-metadata/ocp-metadata.go

```diff
@@ -154,20 +157,24 @@ func getBearerToken(clientset *kubernetes.Clientset) (string, error) {
 	return response.Status.Token, err
 }
 
-// getInfraDetails returns cluster name and platform
-func (meta *Metadata) getInfraDetails() (infraObj, error) {
+// getInfraDetails returns a pointer to an infrastructure object or nil
+func (meta *Metadata) getInfraDetails() (*infraObj, error) {
 	var infraJSON infraObj
 	infra, err := meta.dynamicClient.Resource(schema.GroupVersionResource{
 		Group:    "config.openshift.io",
 		Version:  "v1",
 		Resource: "infrastructures",
 	}).Get(context.TODO(), "cluster", metav1.GetOptions{})
 	if err != nil {
-		return infraJSON, err
+		// If the infrastructure resource is not found we assume this is not an OCP cluster
+		if errors.IsNotFound(err) {
+			return nil, nil
+		}
+		return &infraJSON, err
 	}
 	infraData, _ := infra.MarshalJSON()
 	err = json.Unmarshal(infraData, &infraJSON)
-	return infraJSON, err
+	return &infraJSON, err
```

Comment:
Why are returning this object `by reference` when we are not trying to modify/update it. Can't we just go ahead with return `by value`?

Reply:
Because it's more natural to compare with nil :)